src/main/java/com/minecolonies/colony/buildings/BuildingFarmer.java

```diff
@@ -243,6 +224,29 @@ public void writeToNBT(@NotNull NBTTagCompound compound)
         compound.setBoolean(TAG_ASSIGN_MANUALLY, assignManually);
     }
 
+    @Override
+    public void onDestroyed()
+    {
+        super.onDestroyed();
+        for (@NotNull final Field field : farmerFields)
+        {
+            final Field tempField = getColony().getField(field.getID());
+
+            if (tempField != null)
+            {
+                tempField.setTaken(false);
+                tempField.setOwner("");
+                @NotNull final ScarecrowTileEntity scarecrowTileEntity = (ScarecrowTileEntity) getColony().getWorld().getTileEntity(field.getID());
+                getColony().getWorld()
+                  .notifyBlockUpdate(scarecrowTileEntity.getPos(),
+                    getColony().getWorld().getBlockState(scarecrowTileEntity.getPos()),
+                    getColony().getWorld().getBlockState(scarecrowTileEntity.getPos()),
+                    3);
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Assign this magic number 3 to a well-named constant, and use the constant instead. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS109)